src/openrct2/ride/Ride.cpp

```diff
@@ -4051,17 +4044,19 @@ static sint32 ride_check_for_entrance_exit(sint32 rideIndex)
         if (ride->station_starts[i].xy == RCT_XY8_UNDEFINED)
             continue;
 
-        if (ride->entrances[i].xy != RCT_XY8_UNDEFINED) {
+        if (ride_get_entrance_location_of_station(rideIndex, i).x != LOCATION_NULL) {
```

Comment:
More efficient here to pass "ride" as the first argument.
Similarly on line 4051.